packages/php-wasm/compile/Makefile

```diff
@@ -1,4 +1,30 @@
+# A phony target is one that is not really the name of a file; rather it is
+# just a name for a recipe to be executed when you make an explicit request.
+# There are two reasons to use a phony target: to avoid a conflict with a
+# file of the same name, and to improve performance.
+
+# If you write a rule whose recipe will not create the target file,
+# the recipe will be executed every time the target comes up for remaking.
+# https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html
+.PHONY: all libz libzip libpng16 libxml2 libopenssl libsqlite3 libiconv libncurses libedit bison2.7 oniguruma base-image clean
```

Comment:
do all of these need to be `.PHONY`? I wouldn't ask other than it seems like this patch is now adding that property to the targets, and it wasn't there before.

of course, adding `.PHONY` will ensure they're built every time we run `make`, but particularly for the libraries like `libpng16` and `libzip` it seems like we could be more efficient in general if we based the build on some deterministic element like a version number, build architecture, or source files, etc…

I guess the important question is whether you found something in this PR that was broken because dependencies didn't get rebuilt when they should have. if that's the case then `.PHONY` seems like a fair addition; if not, maybe we let it stay the way it was and address that separately.

Reply:
The Makefile should have originally listed these targets as .PHONY because the names being used are the same as the directories in which the actual artifacts lie. Unintentionally, if that directory has a modification date newer than its dependencies, it would then INCORRECTLY be marked as cached and skipped. Adding the .PHONY target instructs make to skip looking at the directory modification time and instead look at the listed dependencies only.

Reply:
Great. This makes more sense now, as I didn't realize that the directories of the same name as the phony targets existed (because I hadn't looked that deeply outside of the diff).

Let's make sure to add this in the commit description, that this fixes a bug we already had, wherein the `.PHONY` should have always been there.